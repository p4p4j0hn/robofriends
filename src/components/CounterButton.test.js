import React from 'react';
import { shallow } from 'enzyme';
import CounterButton from './CounterButton';

const wrapper = shallow(<CounterButton color='red' />);

describe('counterbutton', () => {
  it('expect to render CounterButton component', () => {
    expect.assertions(1);
    expect(wrapper).toMatchSnapshot();
  });

  it('correctly increments the counter', () => {
    expect.assertions(3);
    expect(wrapper).toMatchSnapshot();
    wrapper.find('[id="counter"]').simulate('click');
    expect(wrapper.state()).toStrictEqual({ count: 1 });
    wrapper.find('[id="counter"]').simulate('click');
    wrapper.find('[id="counter"]').simulate('click');
    expect(wrapper.state()).toStrictEqual({ count: 3 });
  });

  it('only re-renders when required', () => {
    expect.assertions(2);
    expect(wrapper.instance().shouldComponentUpdate({ count: 3 })).toBe(false);
    expect(wrapper.instance().shouldComponentUpdate({ count: 4 })).toBe(true);
  });
});
