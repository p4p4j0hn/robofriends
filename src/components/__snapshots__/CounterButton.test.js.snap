// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`counterbutton correctly increments the counter 1`] = `
<button
  color="red"
  id="counter"
  onClick={[Function]}
>
  Count: 
  0
</button>
`;

exports[`counterbutton expect to render CounterButton component 1`] = `
<button
  color="red"
  id="counter"
  onClick={[Function]}
>
  Count: 
  0
</button>
`;
